native/exe/ServiceControlLaunch/ServiceControlLaunch.cpp

```diff
@@ -126,44 +125,71 @@ int APIENTRY WinMain(HINSTANCE hInstance,
 	DWORD dwRead = 0;
 	DWORD hsize = sizeof(dwRead);
 	hsize = sizeof(currVer);
-	if (RegOpenKeyEx(rootKey, "Software\\JavaSoft\\Java Runtime Environment", 0, KEY_QUERY_VALUE, &myKey) != ERROR_SUCCESS)
+	HMODULE exeMod = GetModuleHandle(NULL);
+	char appPath[512];
```

Comment:
Change the 512 to be _MAX_PATH to be technically correct. :)

Reply:
done